lib/components/Button/Button.tsx

```diff
@@ -1,135 +1,107 @@
 import clsx from 'clsx';
 import React, {
+	AriaAttributes,
+	ButtonHTMLAttributes,
 	cloneElement,
-	ComponentType,
 	createElement,
-	ElementType,
 	forwardRef,
-	FunctionComponent,
 	isValidElement,
+	ReactElement,
 } from 'react';
-import {
-	EProgressSpinnerSize,
-	EProgressSpinnerVariant,
-	ProgressSpinner,
-} from '../ProgressSpinner';
+import { IconType } from '../../icons';
+import { ProgressSpinner } from '../ProgressSpinner';
 
-import { warning } from '@autoguru/utilities';
-import styles from './style.scss';
+import styles from './Button.scss';
 
-export enum ESize {
-	Small = 'small',
+export enum EButtonSize {
 	Medium = 'medium',
+	Small = 'small',
 }
 
-export enum EVariant {
+export enum EButtonVariant {
 	Primary = 'primary',
 	Secondary = 'secondary',
 	Danger = 'danger',
 }
 
-export interface Props {
-	className?: string;
-	component?: ComponentType; // @deprecated
+type ButtonPrimitive = ButtonHTMLAttributes<HTMLButtonElement>;
+
+type AllowedChildren = string | IconType;
+
+interface Props
+	extends Pick<ButtonPrimitive, 'id' | 'onClick' | 'type' | 'className'>,
+		Pick<AriaAttributes, 'aria-label'> {
+	children: AllowedChildren | AllowedChildren[];
 	disabled?: boolean;
-	is?: ElementType;
-	isFullWidth?: boolean;
+	is?: ReactElement;
```

Comment:
Should we limit this to 'a' and 'button' only?

Reply:
This is mainly here to support the `<Link />` react router will give us, etc...